README.md

```diff
@@ -92,19 +105,60 @@ And destructuring tuples with `let`:
 let (x, y) = tup; (+ x y)
 ```
 
-Term duplication is possible using `dup`:
+Term duplication is done automatically when a variable is used more than once. But it's possible to manually duplicate a term using `dup`:
 ```rs
 // the number 2 in church encoding using dup.
 ch2 = λf λx dup f1 f2 = f; (f1 (f2 x))
 
-/// a tagged '#i' dup
-id_id = dup #i id1 id2 = λx x; (id1 id2)
-
 // the number 3 in church encoding using dup.
 ch3 = λf λx dup f0 f1 = f; dup f2 f3 = f0; (f1 (f2 (f3 x)))
 ```
 
-It is possible to use channels, the variable occur outside it's body:
+A `sup` is a superposition of two values, it is defined using curly brackets with two terms inside
+```rs
+sup = {3 7}
+```
+
+Sups can be used anywhere a value is expected, if anything interacts with the superposition, the result is the superposition of that interaction on both the possible values:
+
+```rs
+mul = λa λb (* a b)
+result     = (mul 2 5)         // returns 10
+result_sup = (mul 2 {5 7})     // returns {10 14}
+multi_sup  = (mul {2 3} {5 7}) // returns {{10 14} {15 21}}
+```
+
+To access both values of a superposition, `dups` with labels are needed.  
+A `dup` generally just duplicates the term it points to:
+
+```rs
+// each dup variable now has a copy of the {1 2} superposition
+dup x1 x2 = {1 2}
+```
+
+Both `dups` and `sups` support labels, that is, a field starting with `#` to identify their counterpart:
+```rs
+// here, x1 now contains the value of 1, and x2 the value of 2
+dup #i x1 x2 = {#i 1 2}
+```
+
+Due to how dups are compiled, dup tags between two interacting terms should not contain the same label. For example, an application of the church numeral 2 with itself:
+
+```rs
+c2 = λf λx dup f1 f2 = f; (f1 (f2 x))
+main = (c2 c2)
+```
+
+To avoid label collision, HVM-Lang automatically generates new dup labels for each dup in the code. But with cases like the example above, when the interacting dups comes from the same place, the result is an invalid reduction.  
+
+To fix the problem, its necessary to re-create the term so that a new label is assigned, or manually assign one:
+```rs
+c2  = λf λx dup        f1 f2 = f; (f1 (f2 x))
+c2_ = λf λx dup #label f1 f2 = f; (f1 (f2 x))
+main = (c2 c2_)
+```
+
+It is possible to use channels, the channel variable can occur outside it's body:
```

Comment:
Can you add an explanation on what "channels" are (which I'm not sure if are the name we want to give these). Basically lambdas whose variable escapes their scope. Also, we should probably move this to the advanced section

Reply:
Should we just call then scopeless lambdas?